### PR TITLE
feat: implement Dav for groups

### PR DESCRIPTION
**Pull request by asbiin**
This implements Carddav for groups.

The group type is now optional, carddav doesn't know about the group types and I think it's easier this way - and I think it's better to make it optional anyways.

### CODE REVIEW

File: app/Domains/Contact/ManageContact/Dav/ImportContact.php [hunk 1]
```diff
@@ -114,22 +114,17 @@ private function getExistingContact(VCard $vcard): ?Contact
      */
     private function getContactData(?Contact $contact): array
```
Comment (djaiss) [inline app/Domains/Contact/ManageContact/Dav/ImportContact.php:115]: Why would we call this function without a contact?

**Review by djaiss — COMMENTED** [review]
Good work!

File: app/Domains/Contact/ManageContact/Dav/ImportContact.php (hunk 1 above)
Comment (asbiin) [reply app/Domains/Contact/ManageContact/Dav/ImportContact.php:115]: It happens when we import a new vcard for the 1st time

File: app/Domains/Contact/ManageGroups/Web/Controllers/ContactModuleGroupController.php [hunk 2]
```diff
@@ -50,9 +51,10 @@ public function store(Request $request, string $vaultId, string $contactId): Jso
             'group_type_role_id' => $request->input('group_type_role_id') ?? null,
         ];
 
-        $group = (new AddContactToGroup())->execute($data);
+        AddContactToGroup::dispatchSync($data);
```
[the remark this answers is not in the document]
Comment (asbiin) [reply app/Domains/Contact/ManageGroups/Web/Controllers/ContactModuleGroupController.php:54]: The benefit is: consistency, by always calling `AddContactToGroup` with dispatch. But it's not a big change, true.

**Pull request merged**